libraries/cyclestreets-fragments/src/main/java/net/cyclestreets/CycleMapFragment.kt

```diff
@@ -135,7 +135,7 @@ open class CycleMapFragment : Fragment(), Undoable {
     }
 
     private fun launchFindDialog() {
-        FindPlace.launch(activity, map!!.boundingBox) { place ->
+        FindPlace.launch(context!!, map!!.boundingBox) { place ->
```

Comment:
Do we need the !! here?  Can context or map actually null?

Reply:
@jezhiggins good question; but yes, we do need these.

-  Previously, `FindPlace` was in Java and the `launch` method took a (implicitly nullable) `Context` object.  If it was null (unlikely as that is, clearly) then that would have been a runtime failure.
-  Now, `FindPlace` is in Kotlin, and to avoid peppering that class with `!!` throughout, the `launch` method takes a `Context` object (explicitly non-nullable, as opposed to `Context?` which is explicitly nullable).
-  Since the `Fragment.getContext()` method (and indeed `Fragment.getActivity()` similarly) is annotated as `@Nullable`, then we have to assert here at compile time.

In the case of the `map` property, look at the history of PR #273 - I originally had this as a `lateinit var` (so non-nullable) but found that this caused unnecessary assertion crashes in certain scenarios, so changed it to be nullable.

Reply:
Ah, gotcha.  So we're moving the potential failure site up out of FindPlace to the callsite.  If can, in the future, assert that these things are non-null, the linter will tell and we can clean them up then.